src/hrflow_connectors/core/connector.py

```diff
@@ -763,8 +764,25 @@ class ConnectorModel(BaseModel):
     description: str
     url: str
     type: ConnectorType
+    subtype: t.Optional[str] = Field(
```

Comment:
En fait je pense que le bon typing est `str` et pas `t.Optional[str]`. Avec ce typing dans ton code si tu as un instance d'un connecteur `C = Connector(...)` et que tu fais `C.subtype` ton IDE, outil de static type checking va te donner `str | None` 

Et par exemple tu ne pourras pas faire des trucs du genre `C.subtype + "Some Other String"` il va te donner une erreur du genre `subtype` might be None. Can't add None to string. 

Donc ca va forcer partout a faire des bouts de code comme en dessous alors que en realité tu sais que c'est toujours un string
```py
if C.subtype is not None:
    C.subtype // Ici il te laisse l'utiliser comme string
```

Donc le bon typing il me semble c'est de faire `subtype: str = Field(...)` 

Par contre effectivement dans ton validateur / formatteur `value: str | None` ca c'est une conséquence de `always=True` et `pre=True`. Mais tu vois que ta fonction renvoit toujours `str`

Reply:
C'est pas directement evident de comprendre les impacts de mettre un `Optional` ou `| None` quelque part au début